src/Nethermind/Nethermind.Blockchain/Synchronization/BlockDownloadContext.cs

```diff
@@ -147,8 +147,8 @@ private static void RecoverReceiptData(TxReceipt receipt, Block block, Transacti
             receipt.TxHash = transaction.Hash;
             receipt.Index = transactionIndex;
             receipt.Sender = transaction.SenderAddress;
-            receipt.Recipient = transaction.IsContractCreation ? null : transaction.To;
-            receipt.ContractAddress = transaction.IsContractCreation ? transaction.To : null;
+            receipt.Recipient = transaction.GetRecipient();
+            receipt.ContractAddress = transaction.GetCreatedContract();
```

Comment:
this is actually all incorrect here for created contract - created contract is not the To field